src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -417,5 +414,24 @@ private static string ToBase32StringSuitableForDirName(byte[] buff)
 
             return StringBuilderCache.GetStringAndRelease(sb);
         }
+        
+        // Note: This method returns path.Length for
+        // inputs like "/usr/foo/" on Unix. As such,
+        // it is NOT safe for being used to index
+        // the string without additional verification.
+        private static int FindFileNameIndex(string path)
+        {
+            Debug.Assert(path != null);
+            
+            int length = path.Length;
+            for (int i = length - 1; i >= 0; i--)
+            {
+                char ch = path[i];
+                if (IsDirectoryOrVolumeSeparator(ch))
```

Comment:
Deciding to keep it, because otherwise the line is kind of long (moving this to `PathInternal`, so it looks like `if (Path.IsDirectoryOrVolumeSeparator(path[i]))`.